src/util/postgres_client.js

```diff
@@ -1284,6 +1310,30 @@ class PostgresClient extends EventEmitter {
             if (pg_client) await pg_client.end();
         }
     }
+
+    // Lazy migration of the old mongo style collection/table based
+    // sequences. If a table with the name matching the sequence one
+    // is found, then:
+    // - fetch the current sequence value from the collection
+    // - drop the table
+    // - return the init value to be used for native sequence
+    // If no table is found, return 1 - clean install
+    async migrateFromMongoSequence(name) {
```

Comment:
I think there could be some race conditions here since we have several endpoints starting at once that can run this migration.

Reply:
it does not matter which start value wins the race to `CREATE SEQUENCE IF NOT EXISTS`. we don't care if a couple of seq numbers are skipped. the sequence uniqueness invariant is preserved and one and only  DB sequence instance is created. 

wdyt?

Reply:
Good point @dannyzaken

I have reviewed the sequence creation/migration process. The drop table moved after the native sequence creation completion and `DROP TABLE` became `IF EXISTS`. The drop table becomes reentrant. The possibility of entering the race after the table is dropped but the native sequence does not exist yet and winning by creating a native sequence with init value 1 is excluded. 

So during the endpoints race, whoever wins the following invariants are preserved:
- one and only native sequence is created
- the uniqueness of sequence number in case of migration is preserved.  we don't care if a couple of sequence numbers are skipped, during migration
- in the case of migration the collection table is dropped

Let me know if you still see any issues there.